### PR TITLE
Feature/generate lib

### PR DESCRIPTION
**Pull request by timfrijtersppw**

### CODE REVIEW

File: src/library/schema.json [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+      "$default": {
+        "$source": "argv",
+        "index": 0
+      },
+      "x-prompt": "What name would you like to use for the library?"
+    },
+    "prefix": {
+      "type": "string",
+      "format": "html-selector",
+      "description": "The prefix to apply to generated selectors for the initial project.",
```
Comment (apeeters) [inline src/library/schema.json]: This is not an `initial project` -> `library`.

File: src/library/files/tslint.json.template [hunk 2]
```diff
@@ -10,5 +10,5 @@
     "no-unbound-method": false,
     "no-implicit-dependencies": [true, "dev"]
   },
-  "rulesDirectory": ["node_modules/codelyzer"]
+  "rulesDirectory": ["codelyzer"]
```
Comment (apeeters) [inline src/library/files/tslint.json.template:13]: fixup met 'Add tslint override to generation of library'

File: src/library/index.ts [hunk 3]
```diff
@@ -1,14 +1,21 @@
...
+import { apply, applyTemplates, chain, externalSchematic, mergeWith, Rule, url } from '@angular-devkit/schematics';
+import { MergeStrategy } from '@angular-devkit/schematics/src/tree/interface';
 
 export interface LibraryOptions {
     name?: string;
     prefix?: string,
 }
 
-export default function(options: LibraryOptions): Rule {
+export default function (options: LibraryOptions): Rule {
```
Comment (apeeters) [inline src/library/index.ts:28]: whitespace

File: src/library/index.ts [hunk 4]
```diff
@@ -22,6 +22,7 @@ export interface LibraryOptions {
 
 export default function (options: LibraryOptions): Rule {
     return async (host: Tree) => {
+        const tempOptions = {...options}; // Use the exact same options when using the external schematic, and do what we want with the current one
```
Comment (apeeters) [inline src/library/index.ts]: You probably want cloneDeep here

File: src/library/index.ts [hunk 5]
```diff
@@ -49,7 +51,9 @@ export default function (options: LibraryOptions): Rule {
                     }),
                     move(projectRoot),
                 ]), MergeStrategy.AllowCreationConflict),
-            addDependenciesToPackageJson()
+            addDependenciesToPackageJson(),
+            modifyWorkspace(tempOptions.name),
```
Comment (apeeters) [inline src/library/index.ts]: Why use tempOptions here instead of options?

File: src/library/index.ts [hunk 6]
```diff
@@ -75,4 +79,33 @@ export default function (options: LibraryOptions): Rule {
...
+    }
+
+    function removeConfigurationsFromBuildOptions(project: ProjectDefinition): void {
+        const targets = project.targets;
+        const buildTarget = targets.get('build');
+        if (buildTarget === undefined) {
+            throw new SchematicsException("Architect target missing (build)");
+        }
+        if (buildTarget.configurations === undefined) {
+            throw new SchematicsException("Expected build options to be defined");
```
Comment (apeeters) [inline src/library/index.ts:119]: build options -> build configurations

File: src/library/index.ts [hunk 7]
```diff
@@ -90,6 +91,29 @@ export default function (options: LibraryOptions): Rule {
         });
     }
 
+    function updateAngularCompilerOptionsToTsConfig(folderName: string): Rule {
+        return (host: Tree): Tree => {
+            const path = `${folderName}/tsconfig.lib.json`;
+            const file = host.read(path);
+            // Necessary to remove a comment (comments are not accepted when converting a string to JSON)
```
Comment (apeeters) [inline src/library/index.ts]: Comments in JSON 🤯

Maybe easier to use https://www.npmjs.com/package/strip-json-comments ?
Or maybe even require('typescript').parseConfigFileTextToJson()

Comment (apeeters) [reply src/library/index.ts]: Or better yet: update the config file via typescript api calls.

File: src/library/index.ts (hunk 5 above)
Comment (timfrijtersppw) [reply src/library/index.ts]: Because we want to use the original name here. tempOptions.name = '@coben/common-ui', while options.name = 'common-ui'

Comment (apeeters) [reply src/library/index.ts]: Introduce a descriptive variable for that, then.

**Pull request merged**